rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RMNodeStarter.java

```diff
@@ -1209,17 +1207,32 @@ private ResourceManager loginToResourceManager(final Credentials credentials, fi
      * at the given URL, logs with provided credentials and adds the local node to
      * the Resource Manager. Handles all errors/exceptions.
      */
-    protected ResourceManager registerInRM(final Credentials credentials, final String rmURL, final String nodeName,
-            final Collection<Node> nodes) {
+    protected ResourceManager registerInRM(final Credentials credentials, final String rmURL, String nodeName,
+            Collection<Node> nodes) {
+
+        RMAuthentication rmAuth = joinResourceManager(rmURL);
+        ResourceManager rm = loginToResourceManager(credentials, rmAuth);
 
-        RMAuthentication auth = joinResourceManager(rmURL);
-        final ResourceManager rm = loginToResourceManager(credentials, auth);
-        SigarExposer sigarExposer = null;
+        startMonitoring(rmAuth);
 
+        for (final Node node : nodes) {
+            nodeSetJmxUrl(sigarExposer, node);
+            addNodeToResourceManager(rmURL, node, rm);
```

Comment:
Question: can `addNodeToResourceManager` throw some kind of exception, so we will skip next calls of `addNodeToResourceManager`, if so, maybe, we can catch them?

Reply:
Normally, no, this would be an abnormal behavior. In that case, the global try catch in the main method will exit the JVM with a clear error message.